kit/Kit.cpp

```diff
@@ -3983,4 +4001,225 @@ void dump_kit_state()
+        {
+            bFileNotEnd=false;
+            return false;
+        }
+
+        const StringVector tokens = StringVector::tokenize(line.data(), line.size());
+        if (tokens.size() >= 3)
+        {
+            uint32_t iTime = 0;
+            uint32_t iUser = 0;
+            tokens.getUInt32(0,"time",iTime);
+            tokens.getUInt32(1,"user",iUser);
+            if (tokens.size() > 3)
+                aLoadedLines.emplace_back(iTime, iUser, tokens[2], tokens[3], false);
+            else
+                aLoadedLines.emplace_back(iTime, iUser, tokens[2], "", false);
+            nLoadedLines++;
+            return true;
+        }
+    } while (true);
+}
+
+int LogUiCmd::findNextUserLine(int lineId)
+{
+    int iUserId=aLoadedLines[lineId].iUser;
+    do
+    {
+        lineId++;
+        if (lineId >= nLoadedLines)
+            if (!loadNextLine())
+                return -1;
+        if (iUserId == aLoadedLines[lineId].iUser)
+            return lineId;
+    } while (true);
+}
+
+bool LogUiCmd::mergeIfMouseClick(int lineId)
+{
+    if (aLoadedLines[lineId].aCmd == "cmd:mouse" && aLoadedLines[lineId].aSubCmd == "buttondown")
+    {
+        //find next line from user .. load lines if needed.
+        int nNextLine = findNextUserLine(lineId);
+        if (nNextLine >= 0 && aLoadedLines[nNextLine].aCmd == "cmd:mouse" && aLoadedLines[nNextLine].aSubCmd == "buttonup")
+        {
+            aLoadedLines[lineId].aSubCmd = "click";
+            aLoadedLines[nNextLine].bMerged = true;
+            return true;
+        }
+    }
+    return false;
+}
+
+bool LogUiCmd::removeIfFalseUndoDecrease(int lineId)
+{
+    if (aLoadedLines[lineId].aCmd == "undo-count-change:-1" && aLoadedLines[lineId].aSubCmd == "")
+    {
+        //find next line from user .. load lines if needed.
+        int nNextLine = findNextUserLine(lineId);
+        if (nNextLine >= 0 && aLoadedLines[nNextLine].aCmd == "undo-count-change:+1" && aLoadedLines[nNextLine].aSubCmd == "")
+        {
+            aLoadedLines[lineId].bMerged = true;
+            return true;
+        }
+    }
+    return false;
+}
+
+void LogUiCmd::processLogFile()
+{
+    std::string aTimeLog = "log-start-time: " + aKitStartTimeStr + " user-count:" + std::to_string(aUsersLogged.size());
+    Log::logUI(Log::WRN, aTimeLog.c_str());
+    aFileStreamUICommands.seekg(0, std::ios::beg);
+
+    if (!Log::isLogUIMerged())
+    {
+        // Log the Non-merged file
+        std::string line;
+        while (std::getline(aFileStreamUICommands, line))
+        {
+            if (line.size()>0)
+                Log::logUI(Log::WRN, line);
+        }
+    }
+    else
+    {
+        // Process the UI command log file. merge what we can
+        // and copy it to a bigger permanent file
+        nLoadedLines = 0; //Loaded lined count
+        int nProcessedLines = 0;
+        bFileNotEnd = true;
+        //std::vector<lineOfUiLogFile> aLoadedLines;
```

Comment:
This all looks like core code: we don't use these hungarian notation in online - please fit in with the rest of the code re: coding style.

What is this commented out aLoadedLines ?

Reply:
Changed variable names notations.
the commented code was removed as it was in the post-process